Applications/Custom/LayerClient/jni/pow.h

```diff
@@ -15,89 +15,78 @@
 
   /**
    * @brief Destroy the Pow Layer object
    *
    */
-  ~PowLayer();
+  ~PowLayer() {}
+
+  using nntrainer::Layer::setProperty;
 
   /**
-   * @brief     set Property of layer
+   * @brief     set Property of layer, currently only "exponent is accepted"
    * @param[in] values values of property
    * @retval #ML_ERROR_NONE Successful.
    * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
    */
-  int setProperty(std::vector<std::string> values) {
-    /**< NYI */
-    return 1;
-  }
-
-  /**
-   * @brief Set the Property by propertyType, this is not used in this demo
-   *
-   * @param type property type
-   * @param value value
-   */
-  void setProperty(const ml::train::Layer::PropertyType type,
-                   const std::string &value = "") {
-    /**< NOT USED */
-  }
+  int setProperty(std::vector<std::string> values);
 
   /**
-   * @brief check if hyperparameter is valid
+   * @brief initializing nntrainer
    *
-   * @return int ML_ERROR_NONE if successful
+   * @return int ML_ERROR_NONE if success
    */
-  int checkValidataion() { return ML_ERROR_NONE; }
-
-  /**
-   * @brief Get the Loss bound to the object
-   *
-   * @return float
-   */
-  float getLoss() { return 0.0f; }
+  int initialize();
 
   /**
    * @brief nntrainer forwarding function
    *
    * @param in input tensors
-   * @return nntrainer::sharedConstTensors output tensors
    */
-  nntrainer::sharedConstTensors forwarding(nntrainer::sharedConstTensors in) {
-    return in;
-  }
+  void forwarding(nntrainer::sharedConstTensors in = {});
 
   /**
-   * @brief nntrainer backwaridng function
+   * @brief nntrainer backwarding function
    *
    * @param in input tensors
    * @param iteration number of iterations
-   * @return nntrainer::sharedConstTensors output tensors
+   * @note note that this function is replaced to calcDerivative,
+   * calcGradientis. Please uncomment this function when inheriting from layer.h
+   * Currently, layer_internal has their generic function so this function is
+   * commented out for demonstration purpose.
    */
-  nntrainer::sharedConstTensors backwarding(nntrainer::sharedConstTensors in,
-                                            int iteration) {
-    return in;
-  }
+  // void backwarding(int iteration, nntrainer::sharedConstTensors in = {});
 
   /**
-   * @brief initialize function
-   *
-   * @return int ML_ERROR_NONE if successful
+   * @brief     calc the derivative to be passed to the previous layer
+   * @param[in] in List of Derivative Tensor from the next layer
+   * @retval    Derivative List of Tensor for the previous layer
    */
-  int initialize() { return 1; }
+  void calcDerivative(nntrainer::sharedConstTensors in = {});
+
+  /**
+   * @brief     Calculate the derivative of a layer, in powLayer nothing will be
+   * calculated
+   * @param[in] in List of Derivative Tensor from the next layer
+   */
+  void calcGradient(nntrainer::sharedConstTensors in = {}) {}
```

Comment:
I think this can be ommited.

Reply:
Later if the intention was to give an example with calcGradient, maybe create another example with some weights or ignore this comment otherwise.

Reply:
When creating another example, this could be deleted I guess.

Reply:
Or this example has a weight later on.

Reply:
lets keep this example small as others will be referring and reading this today itself.

Reply:
Done yesterday, calcDerivative, backwarding deleted